### PR TITLE
Using already-defined constants instead of duplicating their values

### PR DESCRIPTION
**Pull request by dvmarcilio**
Hello,

I checked the project on SonarCloud and thought I could fix some issues. 

This PR provides fixes for multiple [String literals should not be duplicated (squid:S1192)](https://rules.sonarsource.com/java/RSPEC-1192) violations (only for already defined constants).

### CODE REVIEW

File: server/sonar-db-dao/src/main/java/org/sonar/db/property/PropertiesDao.java [hunk 1]
```diff
@@ -116,7 +116,7 @@ private static PreparedStatement createStatement(String projectUuid, Collection<
     res.setString(1, projectUuid);
     int index = 2;
     for (String dispatcherKey : dispatcherKeys) {
-      res.setString(index, "notification." + dispatcherKey + ".%");
+      res.setString(index, NOTIFICATION_PREFIX + dispatcherKey + ".%");
```
Comment (henryju) [inline server/sonar-db-dao/src/main/java/org/sonar/db/property/PropertiesDao.java]: :+1:

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v61/RemoveViewsDefinitionFromProperties.java [hunk 2]
```diff
@@ -64,7 +64,7 @@ private void addToInternalProperties(Context context, @Nullable String property)
         " (?,?,?,?)")) {
         long now = system2.now();
         insert
-          .setString(1, "views.def")
+          .setString(1, VIEWS_DEFINITION_PROPERTY_KEY)
```
Comment (henryju) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v61/RemoveViewsDefinitionFromProperties.java]: :+1:

File: server/sonar-server-common/src/main/java/org/sonar/server/platform/UrlSettings.java [hunk 3]
```diff
@@ -68,7 +68,7 @@ public boolean isSecured() {
   private String computeBaseUrl() {
     String host = config.get("sonar.web.host").orElse("");
     int port = config.getInt("sonar.web.port").orElse(0);
-    String context = config.get("sonar.web.context").orElse("");
+    String context = config.get(PROPERTY_CONTEXT).orElse("");
```
Comment (henryju) [inline server/sonar-server-common/src/main/java/org/sonar/server/platform/UrlSettings.java]: :+1:

File: server/sonar-server/src/main/java/org/sonar/server/authentication/ws/ValidateAction.java [hunk 4]
```diff
@@ -59,7 +59,7 @@ public ValidateAction(Configuration config, BasicAuthentication basicAuthenticat
 
   @Override
   public void define(WebService.NewController controller) {
-    controller.createAction("validate")
+    controller.createAction(VALIDATE_ACTION)
```
Comment (henryju) [inline server/sonar-server/src/main/java/org/sonar/server/authentication/ws/ValidateAction.java]: :+1:

File: server/sonar-server/src/main/java/org/sonar/server/platform/ws/UpgradesAction.java [hunk 5]
```diff
@@ -70,7 +70,7 @@ private static void writeMetadata(JsonWriter jsonWriter, Release release) {
 
   @Override
   public void define(WebService.NewController controller) {
-    controller.createAction("upgrades")
+    controller.createAction(ARRAY_UPGRADES)
```
Comment (henryju) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ws/UpgradesAction.java]: The fact two constants have the same value doesn't necessary means we have to use the same constant. Here `ARRAY_UPGRADES` is an element of the JSON response. The fact it has the same value is `coincidental`, and using the same constant make the code less readable by giving the impression the intent was to use the same value in the two places.

File: server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatesAction.java [hunk 6]
```diff
@@ -63,7 +63,7 @@ public UpdatesAction(UserSession userSession, UpdateCenterMatrixFactory updateCe
 
   @Override
   public void define(WebService.NewController controller) {
-    controller.createAction("updates")
+    controller.createAction(ARRAY_UPDATES)
```
Comment (henryju) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatesAction.java]: See my previous comment, should not be the same constant

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuperImpl.java [hunk 7]
```diff
@@ -148,7 +148,7 @@ private QProfileRestoreSummary restore(DbSession dbSession, Reader backup, Funct
       SMInputFactory inputFactory = initStax();
       SMHierarchicCursor rootC = inputFactory.rootElementCursor(backup);
       rootC.advance(); // <profile>
-      if (!"profile".equals(rootC.getLocalName())) {
+      if (!ATTRIBUTE_PROFILE.equals(rootC.getLocalName())) {
```
Comment (henryju) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuperImpl.java]: :+1:

File: server/sonar-server/src/main/java/org/sonar/server/ui/ws/ComponentAction.java [hunk 8]
```diff
@@ -114,7 +114,7 @@ public ComponentAction(DbClient dbClient, PageRepository pageRepository, Resourc
 
   @Override
   public void define(NewController context) {
-    NewAction action = context.createAction("component")
+    NewAction action = context.createAction(PARAM_COMPONENT)
```
Comment (henryju) [inline server/sonar-server/src/main/java/org/sonar/server/ui/ws/ComponentAction.java]: Same as other comments, should not be the same constant

File: server/sonar-server/src/main/java/org/sonar/server/ui/ws/ComponentAction.java [hunk 9]
```diff
@@ -221,7 +221,7 @@ private void writeComponent(JsonWriter json, DbSession session, ComponentDto com
       .prop("isFavorite", isFavourite(session, component));
     String branch = component.getBranch();
     if (branch != null) {
-      json.prop("branch", branch);
+      json.prop(PARAM_BRANCH, branch);
```
Comment (henryju) [inline server/sonar-server/src/main/java/org/sonar/server/ui/ws/ComponentAction.java]: Should not reuse the same constant

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/DeactivateAction.java [hunk 10]
```diff
@@ -74,7 +74,7 @@ public void define(WebService.NewController controller) {
       .setResponseExample(getClass().getResource("deactivate-example.json"))
       .setHandler(this);
 
-    action.createParam("login")
+    action.createParam(PARAM_LOGIN)
```
Comment (henryju) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/DeactivateAction.java]: :+1:

**Comment by dvmarcilio** [comment]
Thanks for the quick review and insights @henryju. I reverted the unwanted changes you pointed.

**Comment by henryju** [comment]
Merged manually in private repo, will be synced soon. Thanks for the contribution.

**Pull request closed without merging**